.github/workflows/docker.yml

```diff
@@ -0,0 +1,83 @@
+name: Build and Push Docker Image to GHCR
+
+on: push
```

Comment:
Does the workflow do anything if it's not on master branch? I would rather limit the branches where it's run here rather than doing it per-step